src/Clapi/TextSerialisation.hs

```diff
@@ -24,10 +30,49 @@ tpBuilder ((Time s f), vs) = tb <> mconcat (map (\v -> (fromChar ' ') <> (cvBuil
   where
     tb = (fss s) <> (fromChar ':') <> (fss f)
 
-encode :: Map.Map Time [ClapiValue] -> Builder
+type TimeSeries = Map.Map Time [ClapiValue]
+
+encode :: TimeSeries -> Builder
 encode ts = header <> bodyBuilder
   where
     -- It is invalid for a time series to be empty, so this use of head is
     -- kinda fine, but the errors will suck:
     header = fromString $ map typeTag $ head $ Map.elems ts
     bodyBuilder = mconcat $ map (\tp -> fromChar '\n' <> tpBuilder tp) (Map.toList ts)
+
+quotedString :: Parser Text
+-- FIXME: escaped quotes unsupported
+quotedString = char '"' >> takeTill (== '"')
+
+cvParser :: Char -> Parser ClapiValue
+cvParser 'B' = (wordMatch 'T' True) <|> (wordMatch 'F' False)
+  where
+    wordMatch c v = char c >> return (ClBool v)
+cvParser 'i' = decimal >>= return . ClInt32
+cvParser 's' = ClString <$> quotedString
+
+getTupleParser :: Parser (Parser [ClapiValue])
+getTupleParser = sequence <$> many1 (satisfy (inClass "Bis") >>= \c -> return (char ' ' >> cvParser c))
```

Comment:
Might want to add a `FIXME` to say about generating the type tag characters from somewhere? It might be a little harder given you've defined a slightly altered set from the binary serialisation...